src/ipc/ipc4/helper.c

```diff
@@ -627,6 +627,11 @@ int ipc4_create_chain_dma(struct ipc *ipc, struct ipc4_chain_dma *cdma)
 
 	ipc_pipe = ipc_get_comp_by_id(ipc, pipeline_id);
 
+	if (!ipc_pipe) {
```

Comment:
this too is not correct. The pipeline is create just above. If it wasnt created successfully, we'd never get here right?

Reply:
Yes, I missed the call inside ipc4_create_pipeline() function. Reverting back.